src/TheToothpickGame.java

```diff
@@ -159,146 +292,274 @@ else if (compSkillLevel == HARD)
      *         also NOT be greater than the number of toothpicks left in the pile.
      */
-    
-    
+    private static int getRandomChoice(){
+        int numToothpicks = (int)(Math.random() * maxToothpicksPerTurn ) + 1;
+        while(numToothpicks > toothpicksRemaining){
+            numToothpicks = (int)(Math.random() * maxToothpicksPerTurn ) + 1;
+        }
+        return numToothpicks;
+    }
+
     /**
      * The getOptimalChoice method will return the number of toothpicks that
      * needs to be taken to force a win, if it exists.
-     * 
+     *
      * @return an integer representing the optimal number of toothpicks to take
      *         in order to force a win, if it exists;
      *         this will return -1 if it is impossible to force a win at this time
      */
-    
-    
+    private static int getOptimalChoice(){
+        int numToothpicks;
+        if(toothpicksRemaining % (maxToothpicksPerTurn + 1) != 0){
+            numToothpicks = toothpicksRemaining % (maxToothpicksPerTurn + 1);
+        }else{
+            numToothpicks = getRandomChoice();
+        }
+        return numToothpicks;
+    }
+
     /**
-     * The easyComputerTakesTurn method will always take a random number of 
+     * The easyComputerTakesTurn method will always take a random number of
      * toothpicks and then output their choice to the screen after calling
      * getEasyTurnDescription.
-     * 
+     *
      * Reminder!  In the The Toothpick Game.pdf file, I told you this method
      * (and the next five methods after it!!) should not be using any global
      * variables.  Only local is needed due to the paramater lists.
      */
-    
-    
+
+    private static void easyComputerTakesTurn(){
+        int numToothpicks = getRandomChoice();
+        toothpicksRemaining -= numToothpicks;
+        System.out.println(getEasyTurnDescription(numToothpicks));
+    }
     /**
      * getEasyTurnDescription will return a randomly selected String showing
-     * what the Easy Computer decided to do this turn.  The method will be 
+     * what the Easy Computer decided to do this turn.  The method will be
      * randomly choosing from at least four different possible Strings.
-     * 
-     * The String returned will reflect the easy computer's personality. 
-     * 
+     *
+     * The String returned will reflect the easy computer's personality.
+     *
      * @param  num  an integer representing the number of toothpicks which
      *              is supposed to be taken this turn
      * @return a String which describes what the easy computer does this turn
      */
-    
-    
+    private static String getEasyTurnDescription(int num){
+        int randomNum = (int)(Math.random() * 4);
+        String output = "EasyComputer: ";
+        if(randomNum == 0)
+            output += "Hello, my name is Intelligence Artificial. I take " + num + " this turn! Heyhey, are you excited?";
```

Comment:
Computer descriptions
Note the spec says "Notice that in these outputs, the String returned DOES take into account the proper plural for toothpick
every time. You need to do that also."
Make sure to use "toothpick" if num == 1 and "toothpicks" otherwise.